### ISSUE TITLE
Add requirements.txt and environment.yml for dependencies

### ISSUE THREAD

**Issue by lilyminium** [MDAnalysis/mdanalysis#2543]
**Is your feature request related to a problem? Please describe.**
Potential contributors are recommended to [install MDAnalysis dependencies themselves](https://www.mdanalysis.org/UserGuide/contributing_code.html#building-mdanalysis) before building the package. This is quite long and we are likely to forget to update the user guide every time the dependencies are updated.

**Describe the solution you'd like**
Maintain a `requirements.txt` and `environment.yml` file with dependencies instead. `environment.yml` has the additional benefit of allowing you to specify a prefix, so troubleshooting could begin with "have you run `conda activate mda-dev`"? Also, you can specify pip packages to be installed, which helps with packages like `parmed` that don't have conda builds for Python < 3.7.

**Describe alternatives you've considered**

**Additional context**
Current list of dependencies:

```bash
conda install -c biobuilds -c conda-forge \
 cython numpy mmtf-python mock six biopython \
 networkx cython matplotlib scipy griddataformats \
 hypothesis gsd codecov "seaborn>=0.7.0,<=0.9" \
 clustalw=2.1 netcdf4 scikit-learn "joblib>=0.12" \
 psutil pytest
```

**Comment by richardjgowers** [comment]
This would definitely be cool, especially if our CI tools also use this to avoid duplication etc

**Comment by lilyminium** [comment]
Yep. The user guide already has [requirements.txt](https://github.com/MDAnalysis/UserGuide/blob/master/requirements.txt) and [environment.yml](https://github.com/MDAnalysis/UserGuide/blob/master/environment.yml) although it only uses environment.yml in [Travis](https://github.com/MDAnalysis/UserGuide/blob/master/.travis.yml)

**Comment by tylerjereddy** [comment]
I'd suggest also looking at the modern `pyproject.toml`, based on [PEP 518](https://www.python.org/dev/peps/pep-0518/), which NumPy & SciPy have adopted.

For some context, enabling build isolation via pip can be quite useful: https://github.com/scipy/scipy/issues/8734

**Comment by hmacdope** [comment]
This can be closed now IMHO? @lilyminium feel free to reopen if you feel differently?

**Comment by IAlibay** [comment]
Re-opening, this is not a solved issue @hmacdope

**Comment by hmacdope** [comment]
My bad
